### DISCUSSION TITLE
Parametrierung weiterer Werte von Sensoren/Berechnungen

### DISCUSSION THREAD

**Discussion by hoetzgit** [Daveiano/weewx-wdc#84] [Q&A]
Hallo David,

ist es möglich, weitere Werte von Sensoren oder Berechnungen - die nicht standardmäßig im WeeWX Schema enthalten sind - zu parametrieren? Für das Label sollte das ja funktionieren, aber z.B. für das Icon ~~oder die Farbe~~?

Vielen Dank für Deine tolle Arbeit und Grüße
Henry

**Comment by Daveiano** [comment]
Guter Punkt - zur Zeit ist es so, das es für einige vordefinierte Observations die entsprechenden Icons im Skin gibt, siehe https://github.com/Daveiano/weewx-wdc/blob/v2.3.0/bin/user/weewx_wdc.py#L58

```
    @staticmethod
    def get_icon(observation):
        """
        Returns an include path for an icon based on the observation
        @see http://weewx.com/docs/sle.html
        Args:
            observation (string): The observation
        Returns:
            str: An icon include path
        """
        icon_path = "includes/icons/"

        if observation == "outTemp" or observation == "inTemp":
            return icon_path + "temp.svg"

        elif observation == "outHumidity" or observation == "inHumidity":
            return icon_path + "humidity.svg"

        elif (
                observation == "barometer"
                or observation == "pressure"
                or observation == "altimeter"
        ):
            return icon_path + "barometer.svg"

        elif observation == "windSpeed":
            return icon_path + "wind-speed.svg"

        elif observation == "windGust":
            return icon_path + "wind-gust.svg"
...
```

Besser wäre es natürlich, der Nutzer könnte selbst entscheiden welche Icons er nutzen will. Die bisherige Logik könnte man ja als Fallback belassen?

Außerdem würde ich in Zuge dessen allen im Standard schema enthaltenen observations ein Icon zuweisen, einige fehlen da nämlich noch, also haben noch kein Icon (z.B. co, lightning_strike, no2, windrun).

**Reply by hoetzgit** [reply]
Ja genau, die Logik mind. als Fallback behalten, da sie sicherstellt, dass die Funktionalitäten erhalten bleiben, die Du als Skin Ersteller vorgesehen hast. Sie würde nur durch die Möglichkeit des Überschreibens oder Ergänzens erweitert (bei get_color machst Du das ja auch bereits). Auch könnten Fehlkonfigurationen des Users, .z.B. bei Nichtvorhandensein eines Icons abgefangen und durch defaults ersetzt werden.  Mir fallen da auch noch binding, units, rounding, Erweiterungen in get_show_min/max/sum u.v.m. ein. 

Die Erweiterung der Icons auf die im Standard Schema enthaltenen Werte ist in jedem Fall sinnvoll. Hoffentlich findest Du für alle Werte Icons ;-)

Henry

**Reply by Daveiano** [reply]
Ja das passt gut das du das ansprichst. 

Das [3.0.0](https://github.com/Daveiano/weewx-wdc/milestone/12) Release ist gerade in der Planung. Dieses wird mehrere größere Anpassungen beeinhalten die die vom Nutzer anpassbaren Möglichkeiten stark erweitern werden. 

- Custom **Data bindings** sind schon ein Thema, siehe #71 
- **Units** können doch über die allgemeine weewx.conf gesetzt werden pro Observation oder was meinst du?
- **Rounding** wäre auch ein Thema
- Auch **show_min, show_max oder show_sum** könnte man konfigurierbar machen

Also: gerne her mit Ideen was noch alles anpassbar sein könnte/sollte!

**Reply by hoetzgit** [reply]
Mit "Units" ein kleines Beispiel. Ich habe den Wert Sonnenscheindauer sunshineDur in der DB in **Sekunden**. In einem Tagesdiagramm möchte ich die Sonnenscheindauer sunshineDur in **Minuten** (gruppiert nach Stunde) anzeigen, in einem Monatsdiagramm sunshineDur in **Stunden** (gruppiert nach Tag). Oder funktioniert das schon und ich hab's übersehen?

**Reply by hoetzgit** [reply]
jetzt wo ich es gerade sehe, vielleicht auch die Deaktivierung eines Icons, bspw. beim Diagramm "Temperatur Min/Max/Avg". Da reicht eigentlich auch ein Icon statt bisher drei. `icon=none` oder `icon=`

**Reply by Daveiano** [reply]
> Mit "Units" ein kleines Beispiel. Ich habe den Wert Sonnenscheindauer sunshineDur in der DB in **Sekunden**. In einem Tagesdiagramm möchte ich die Sonnenscheindauer sunshineDur in **Minuten** (gruppiert nach Stunde) anzeigen, in einem Monatsdiagramm sunshineDur in **Stunden** (gruppiert nach Tag). Oder funktioniert das schon und ich hab's übersehen?

Kann ich dir gar nicht spontan sagen, weil ich keine observation mit Zeit-Unit habe. Der Fall ist ja ziemlich spezial, weil alle anderen Observations ändern ja ihre Einheit in der Regel nicht. `sunshineDur` ist ja nicht im Standard weewx Schema enthalten, von welcher Extension kommt das bei dir? Vielleicht kannst du mir zu Testzwecken deine weewx Datenbank zur Verfügung stellen?

> jetzt wo ich es gerade sehe, vielleicht auch die Deaktivierung eines Icons, bspw. beim Diagramm "Temperatur Min/Max/Avg". Da reicht eigentlich auch ein Icon statt bisher drei. icon=none oder icon=

Ja, das sehe ich eher als Bug. Es werden bei kombinierten Diagrammen immer die Icons der jeweiligen Observations gezeigt. Wenn die Observation aber mehrfach im Diagramm auftaucht (z. B. bei outTemp min/max und avg), sollte das Icon nur einmal gezeigt werden. 
Aber ja, sobald die Icons vom Nutzer konfigurierbar sind, kann er das dann sowieso auch ändern.

**Reply by hoetzgit** [reply]
Das mit den Units ist auch nicht auf "meinem Mist" gewachsen, aber inzwischen finde ich es sehr sinnvoll. z.B. immer dann, wenn man kleine Einheiten hat und möchte sie in größeren Einheiten für längere Zeiträume darstellen. Gerne kann ich Dir einen Export meiner DB geben, Du könntest mir nur sagen, wo ich sie für Dich am besten ablegen kann, Cloud etc.

Hier habe ich den Änderungswunsch das erste Mal gesehen:
https://github.com/poblabs/weewx-belchertown/pull/773

Henry

**Edit**: `sunshineDur` kommt ursprünglich als `sunshine_time` von hier: https://github.com/Jterrettaz/sunduration
Da ich mich aber offensichtlich nie mit dem Standard zufrieden geben kann und gerne "rumprobiere", habe ich es für mich auf Sekunden (WeeWX unterstützt ja `sunshineDur : group_deltatime`) angepasst.

**Reply by Daveiano** [reply]
> Das mit den Units ist auch nicht auf "meinem Mist" gewachsen, aber inzwischen finde ich es sehr sinnvoll. z.B. immer dann, wenn man kleine Einheiten hat und möchte sie in größeren Einheiten für längere Zeiträume darstellen. Gerne kann ich Dir einen Export meiner DB geben, Du könntest mir nur sagen, wo ich sie für Dich am besten ablegen kann, Cloud etc.
> 
> Hier habe ich den Änderungswunsch das erste Mal gesehen: [poblabs/weewx-belchertown#773](https://github.com/poblabs/weewx-belchertown/pull/773)

Ich verstehe, ich behalte es im Hinterkopf für die [3.0.0](https://github.com/Daveiano/weewx-wdc/milestone/12). Die Datenbank kannst du mir gerne per https://wetransfer.com/ oder einfach auch per Mail, je nach Größe senden: david.baetge@gmail.com

> **Edit**: `sunshineDur` kommt ursprünglich als `sunshine_time` von hier: https://github.com/Jterrettaz/sunduration Da ich mich aber offensichtlich nie mit dem Standard zufrieden geben kann und gerne "rumprobiere", habe ich es für mich auf Sekunden (WeeWX unterstützt ja `sunshineDur : group_deltatime`) angepasst.

Ah ja die extension habe ich schon mal gesehen, leider aber nur für Davis Vantage Pro Stationen?

**Comment by Daveiano** [comment]
Habe mal einen Schwung issues aus den Ideen hier angelegt:

#87 #88 und #89

Weitere Ideen/Vorschläge sind immer willkommen!
